src/spatial_ref/tests.rs

```diff
@@ -245,6 +245,25 @@ fn failing_transformation() {
     }
 }
 
+#[test]
+#[cfg(all(major_ge_3, minor_ge_3))]
+fn invalid_transformation() {
+    use super::srs::CoordTransformOptions;
+
+    // This transformation can be constructed only if we allow ballpark transformations (enabled by
+    // default).
+    let ma = SpatialRef::from_epsg(6491).unwrap(); // Massachusetts
+    let nl = SpatialRef::from_epsg(28992).unwrap(); // Netherlands
+    let trafo = CoordTransform::new(&ma, &nl);
+    assert!(trafo.is_ok());
+
+    let mut options = CoordTransformOptions::new().unwrap();
+    options.set_ballpark_allowed(false).unwrap();
+    let trafo = CoordTransform::new_ex(&ma, &nl, &options);
+    let err = trafo.unwrap_err();
+    assert!(matches!(err, GdalError::NullPointer { .. }), "{:?}", err);
+}
+
```

Comment:
Would it be viable to provide a test for `set_coordinate_operation`?

Reply:
If somebody could tell me input and expected output, sure. I only implemented it for completeness. If it's not cool to have untested (but also rarely used!) code in there, I'm also happy to remove it again.

Reply:
I think we could call it just to check it doesn't crash.

Reply:
Added a test.